cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -236,6 +234,7 @@ protected void rebalanceForBroker(Broker broker, ClusterModel clusterModel, Set<
                                                        .collect(Collectors.toSet());
       if (maybeApplyBalancingAction(clusterModel, replica, eligibleBrokers, ActionType.INTER_BROKER_REPLICA_MOVEMENT, optimizedGoals,
                                     optimizationOptions) == null) {
+        _provisionResponse = new ProvisionResponse(ProvisionStatus.UNDER_PROVISIONED);
```

Comment:
Why do we need this? Isn't this handled in `AbstractGoal#optimize`?

Reply:
Got it. I just noticed it.